tests/integration/suite/test_catalog.py

```diff
@@ -313,11 +313,52 @@ def get_errored_catalog(catalog):
     assert "areallylongnamelikereallyreallylongwestillneedmorez234dasdfasd"\
         not in templates
     assert "bad-chart_name" not in templates
-    assert catalog.state == "refreshed"
-    assert catalog.transitioningMessage == "Error syncing catalog " + name
+    assert catalog.state == "processed"
+    assert catalog.transitioning == "error"
+    assert "Error syncing catalog" in catalog.transitioningMessage
+    assert "areallylongnamelikereallyreallylongwestillneedmorez234dasdfasd" \
+        in catalog.transitioningMessage
+    assert "bad-chart_name" in catalog.transitioningMessage
     # this will break if github repo changes
     assert len(templates) == 6
     # checking that the errored catalog can be deleted successfully
     client.delete(catalog)
     wait_for_template_to_be_deleted(client, name)
     assert not client.list_catalog(name=name).data
+
+
+def test_invalid_catalog_chart_urls(admin_mc, remove_resource):
+    """Test chart with file:// and local:// url paths"""
+    client = admin_mc.client
+    name = random_str()
+    url = "https://github.com/brendarearden/integration-test-charts"
+    catalog = client.create_catalog(name=name,
+                                    branch="invalid-urls",
+                                    url=url,
+                                    )
+    remove_resource(catalog)
+    wait_for_template_to_be_created(client, catalog.id)
+
+    def get_errored_catalog(catalog):
+        catalog = client.reload(catalog)
+        if catalog.transitioning == "error":
+            return catalog
+        return None
+    catalog = wait_for(lambda: get_errored_catalog(catalog),
+                       fail_handler=lambda:
+                       "catalog was not found in error state")
+    templates = client.list_template(catalogId=catalog.id).data
+
+    assert "aks-goodbyeworld" not in templates
```

Comment:
Is this a bad url in the chart ?

Reply:
It is a bard url in the index.yaml of the chart repository - so its the url the deploy app is going to use to get the chart files - I added comments to help a little